### ISSUE TITLE
FEATURE: Single flag for use ida for ALL (symbolizer, rooter, loader)

### ISSUE THREAD

**Issue by RabidCicada** [BinaryAnalysisPlatform/bap#685]
Just to simplify command line.

Its my naive experience so far that really you need IDA to get anywhere with analysis for even simple linux executables like less.

Something like --ida that just puts ida in for all the configurable flags.

**Comment by ivg** [comment]
So, if IDA installed, in the latest version it will be used both for symbolizer and rooter automatically. In fact, all sources will be used and the information is merged. So there is no need to specify a symbolizer or rooter on the command line. 

The loader is a different story, currently the support for the IDA loader is rather rudimentary, we are hoping to make it better in future releases. Then we can use ida loader by default if it is present.

**Comment by ivg** [comment]
Due to the de-centralized structure of the BAP platform this won't be possible in general. 

1) For example, later we can later implement abi support using IDA, so we will need to update this `--ida` switch to handle the abi also. This will contradict with the open to extension/closed to modification principle

2) Same as previous, but what if we support an abi only for a specific target (substitute abi to rooter/loader/whatever), so "always" is not always an option even for IDA

3) BAP is a component based framework, and we are trying to keep all components independent on each other. That means, that the components should operate only on their ground and do not dictate other components what to do. To put it more simple, suppose we decided to add --radare2 option with the same behavior, how we should resolve a possible conflict between radare2 and IDA if both `--ida` and `--radare2` options are specified? Another example, what if `--rooter=objdump` and `--ida` are specified, should the latter option cancel the first? So it will eventually lead to more confusion. 

On the other hand, I agree that it might be tedious to specify all this parameters every time you run bap. For that we have two other ways to specify parameters: a config file and environment variables. 
Basically, for each option there is the following chain of possible values, with latter overriding the former: 
1. hardcoded default
2. configuration file
3. environment variable 
4. explicit command line arguments

I will explain later in documentation how to use this facility :)
